src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -364,16 +372,13 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
 
   Forward(predictors.cols(begin, begin + batchSize - 1));
   double res = outputLayer.Forward(
-      boost::apply_visitor(outputParameterVisitor, network.back()),
+      network.back()->OutputParameter(),
       responses.cols(begin, begin + batchSize - 1));
 
   for (size_t i = 0; i < network.size(); ++i)
-  {
-    res += boost::apply_visitor(lossVisitor, network[i]);
-  }
+    res += network[i]->Loss();
```

Comment:
maybe this should be: res += LossUpdate(network[i])?

Reply:
I think this is correct the way it is.